Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/performance/AzurePerformanceMetricsCollector.java

```diff
@@ -0,0 +1,138 @@
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.performance;
+
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.io.FileUtils;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperationRef;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperationUtils;
+import com.microsoft.azure.toolkit.lib.common.operation.IAzureOperation;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskContext;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.telemetry.AppInsightsClient;
+import lombok.SneakyThrows;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class AzurePerformanceMetricsCollector {
+    private static final File file = new File(System.getProperty("user.home") + "/performance.csv");
+
+    private static final String PERFORMANCE = "PERFORMANCE";
+    private static final String TELEMETRY_OP_TIMESTAMP = "timestamp";
+    private static final String TELEMETRY_OP_ACTION = "action";
+    private static final String TELEMETRY_OP_CONTEXT_ID = "context_id";
+    private static final String TELEMETRY_OP_ID = "id";
+    private static final String TELEMETRY_OP_PARENT_ID = "parent_id";
+    private static final String TELEMETRY_OP_NAME = "name";
+    private static final String TELEMETRY_OP_TYPE = "type";
+    private static final String OP_ACTION_CREATE = "CREATE";
+    private static final String OP_ACTION_ENTER = "ENTER";
+    private static final String OP_ACTION_EXIT = "EXIT";
+
+    public static void afterCreate(final IAzureOperation op) {
+        final long timestamp = System.currentTimeMillis();
+        final Map<String, String> properties = buildProperties(op);
```

Comment:
duplicate exception:        properties.put(TELEMETRY_OP_ACTION, OP_ACTION_CREATE);
please wrap it in a private common method

Reply:
I don't understand.